Adding Gulp

bakesteve opened the pull request:
The makefiles included in the project dont work on windows 8, using [GnuMake](http://gnuwin32.sourceforge.net/packages/make.htm)

Added in a simple gulp based system (though left make there) to:
- browserify with reactify transform
- compile less
- compile examples
- watch
- using [browserSync](http://browsersync.io) launch a local server

It should make it easier for people to contribute / use this on windows as it builds up and runs a local site with some examples

Havent looked at replacing the standalone or docs makefiles, but would be much appreciated by those of us not on *nix if this used gulp (or grunt)...
For the standalone part, the issue was with browserify_shim as it isnt compatible with browserify 5+ so would have to have a separate gulpfile under standalone (like you do for makes)

How would you feel about using webpack (with a pull request, of course)?

Commit:
gulp working
using tasks folder to split up
has steps for:
- building js (using browserify)
- watching js (using browserify - watchify not working, browserify version issue?)
- browser sync and a server
- less

Commit:
working examples folder
- include any example in the components array
- will get a menu command and load on click
- loads in bootstrap and the grids styles
- should work on any platform as its all gulp'ed

Commit:
ignore any build folder

Commit:
update readme and comments on how to add examples

Commit:
remove build files (should have been gitignored?

andreypopp commented on gulpfile.js:
```diff
@@ -0,0 +1,90 @@
+
+require('./gulp');
+
+// var gulp = require('gulp');
```
Why is the code commented?

andreypopp commented on themes/bootstrap.less:
```diff
@@ -1,3 +1,4 @@
+@import "node_modules/bootstrap/less/bootstrap.less";
```
This would be incompatible with other build setups which include bootstrap from different locations. I'd recommend removing that line and instead consider this less file to be dependent on bootstrap implicitly.

bakesteve replied:
good point.
I tend to prefer avoiding implicit dependencies, but will move this into the examples folder so its only there we are explicitly assuming you get BS3 via npm (not bower, etc)

andreypopp replied:
Yeah, me too, but unfortunately styling is broken on the web :(

Though I hope something like [this](https://github.com/andreypopp/IntegratedCSS/blob/master/example/Button.js) will improve the state of art.

Commit:
Gardening

andreypopp commented:
Great PR!

I wonder if you'd be ok to align code to the codestyle in this project — 2 spaces per indent, space between `if` and `(` and so on. What do you think?

bakesteve commented:
Course – I’m using atom (and new-ish to it) but sure I can set that up to be the defaults

Commit:
Move imports into a seppcific examples.less file, play nice woth other build systems

Commit:
clean out tabs

Commit:
fix spacing for if(s)

bakesteve commented:
@andreypopp think I caught the styling issues, but shout if I missed anything
For the .less import, created a examples.less that imports both the standard, and grid bootstrap.less, so should be ok?

Commit:
- Fix examples css reference
- add a gulp clean task to make sure we always generate the latest files
- dont require addons where we are not using them

andreypopp commented:
@bakesteve hey, sorry for not merging in your PRs. They look great and I definitely plan to merge them soon. This is just because we have a lot of stuff to work on internally. Thanks!

bakesteve commented:
ok - good stuff.
be good to chat at some stage about where this project is going and if / how you'd like contributions

we've now got working examples, based off your selectableCells example with different editors (text boxes, drop downs, autocompletes), keyboard navigation, drag down and copy & paste and would be good to know:
1. what your roadmap is for this, ideally to avoid duplication of effort
2. how you see the structure for this working, as these features all feel like add-ons (or, despite the jquery-nightmare connotations, plugins..)
3. any plans for a comprehensive test suite, and saucelabs integration
4. now you have moved to sphinx, any plans to move the docs -> readthedocs with a webhook? (partially due to makefile woes on windows)

We'd be happy to contribute on most of these, so if you are happy for that, maybe setup a chat at some point (skype, etc)?

andreypopp commented:
Hi, Steven, we should definitely chat about react-grid. We have an important deadline next week, and after that I'll get back to you. Thanks! Sorry for being so unresponsive to your contributions so far :(

bakesteve commented:
No problem – know how it gets when you get those sort of deadlines!
